### ISSUE TITLE
array_io: _write_chunks fails and runs into RecursionLimit when chunksize is (1,)

### ISSUE THREAD

**Issue by Manangka** [Deltares/imod-python#105]
In GitLab by @tessodk on Nov 13, 2019, 14:24

The issue is that a slice(0, 1) returns an identical object:

```python
        for chunksize in chunksizes:
            end = start + chunksize
            b = a.isel({dim: slice(start, end)})
            # Recurse
            _write_chunks(b, pattern, d, nodata, dtype, write)
```

```python
a.identical(b) == True
```

Maybe easiest to special case it?

```python
    if nochunks or dim_is_xy or chunksizes == (1,):  # Base case
```

**Comment by Manangka** [comment]
In GitLab by @visr on Nov 13, 2019, 17:16

Hmm, isn't that already caught by the second half of `nochunks = a.chunks is None or max(map(len, a.chunks)) == 1`? What does that return then in this case?

**Comment by Manangka** [comment]
In GitLab by @Huite on Dec 9, 2019, 19:05

The issue was that `a.isel({dim: slice(start, end)})` does nothing in reducing the dimensions with chunks. It will succesfully select a single chunk in this dimension, but if there is a second dimension with chunks, these will obviously still exist.

Concretely, say there's dims `time, layer, y, x`, with chunks:

`((1, 1), (1, 1, 1), (1300,) (1200,))`

The first isel will select have this as a result:

`((1,), (1, 1, 1), (1300,) (1200,))`

The first chunk, time, still exists, and `max(map(len, a.chunks)) == 1` will evaluate False (since the biggest chunks has length 3)).


The solution lies in jumping to the next dimension, once only a single chunksize is present.

```python
        for dim, chunksizes in zip(a.dims, a.chunks):
            if len(chunksizes) > 1:
                break
```

Note that by necessity, len(chunksizes) > 1 must exist, that's what max(map(len, a.chunks)) == 1 tests for.

**Comment by Manangka** [comment]
In GitLab by @Huite on Dec 9, 2019, 19:06

closed via commit 6c1e946988ca75ec0a65d3c532f63e9906459738

**Comment by Manangka** [comment]
In GitLab by @Huite on Dec 15, 2019, 16:52

mentioned in commit 6c1e946988ca75ec0a65d3c532f63e9906459738
